iree/turbine/kernel/wave/cache.py

```diff
@@ -206,6 +223,13 @@ def store_kernel_to_file(
         _write_file(cur_module_path, "w", module_str)
         kernel_sig_str = json.dumps([usage.name for usage in kernel_sig])
         _write_file(cur_kernelsig_path, "w", kernel_sig_str)
+        cur_hsaco_path = glob.glob(str(WAVE_RUNTIME_DIR / "*.hsaco"))
+        # Copy the hsaco file to the cache directory only if it exists.
+        if cur_hsaco_path:
+            cur_hsaco_path = cur_hsaco_path[0]
+            shutil.copy(cur_hsaco_path, cur_cache_basefile.with_suffix(".hsaco"))
+        cur_pkl_path = cur_cache_basefile.with_suffix(".pkl")
```

Comment:
I have a some questions on this:
1. Do we expect the directory/file to exist because the one who generated it will be the `dump_binaries` config?
2. Why always choose the first file found? `cur_hsaco_path[0]`, shouldn't we match to hash name?

Reply:
Yes, when the wave runtime is enabled, then we do expect the file to exist because it will be generated by iree because we pass in the dump_binaries flag. We can always choose the first file found because we are maintaining the invariant that at any point of time they will always be only one hsaco file in the wave runtime directory.